pkg/dashboards/dashboards.go

```diff
@@ -845,3 +730,66 @@ func ProcessDeleteDashboardsByOrgId(orgid uint64) error {
 	}
 	return nil
 }
+
+func getDashboardsWithMetadata(orgid uint64) (map[string]interface{}, error) {
+	dashboardIds, err := getAllDashboardIds(orgid)
+	if err != nil {
+		return nil, err
+	}
+
+	dashboardDetails := make(map[string]interface{})
+	for id, name := range dashboardIds {
+		dashboardInfo, err := getDashboardMetadata(id, name)
+		if err != nil {
+			log.Errorf("Error getting metadata for dashboard %s: %v", id, err)
+			continue
+		}
+		dashboardDetails[id] = dashboardInfo
+	}
+
+	return dashboardDetails, nil
+}
+
+func getDashboardMetadata(id string, name string) (map[string]interface{}, error) {
+	dashboardInfo := map[string]interface{}{
+		"name":       name,
+		"isFavorite": false,
+		"isDefault":  isDefaultDashboard(id),
+	}
```

Comment:
Same here: it would be good to make this a struct instead of a `map[string]interface{}`